### PR TITLE
Addon-contexts: Remove peer dependencies

### PR DESCRIPTION
**Pull request by leoyli**
Issue: #7625

## What I did

Remove `peerDependencies` since they install large unnecessary packages into your project.

## How to test

N/A

### CODE REVIEW

**Review by shilman — APPROVED** [review]
LGTM!

**Comment by Sjiep** [comment]
@leoyli any idea how to fix the checks?

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
Fixed the checks, I think

File: addons/contexts/src/manager/ContextsManager.tsx [hunk 1]
```diff
@@ -22,11 +22,13 @@ export const ContextsManager: ContextsManager = ({ api }) => {
   );
 
   // from preview
-  useChannel(UPDATE_MANAGER, newNodes => setNodes(newNodes || []), []);
+  const emit = useChannel({
+    [UPDATE_MANAGER]: newNodes => setNodes(newNodes || []),
+  });
```
Comment (leoyli) [inline addons/contexts/src/manager/ContextsManager.tsx]: This is pretty sweet <3

**Comment by leoyli** [comment]
Maybe we can merge this now?!

**Comment by shilman** [comment]
@leoyli we'll get it first thing in 5.3

**Comment by shilman** [comment]
@leoyli Can you also do this for `addon-centered`: #7989

**Comment by leoyli** [comment]
@shilman, sure things, can you create an issue later for me and I will find a weekend to finish things up?! I'm really want to add more value back to the community but I need a bit momentum/pressure :p.

**Comment by leoyli** [comment]
@shilman, Is anything we want to do here to merge this PR?  I guess you have a script to update package in the monorepo, so I think I should not do this by hand :p

**Comment by shilman** [comment]
thanks for the ping @leoyli - will merge once CI succeeds

**Comment by leoyli** [comment]
All green! Let's merge it :D

**Pull request merged**